src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -672,10 +672,16 @@ protected function dumpContainer(ConfigCache $cache, ContainerBuilder $container
      *
      * @param ContainerInterface $container The service container
      *
+     * @throws \DomainException If the container is not of type ContainerBuilder
+     *
      * @return DelegatingLoader The loader
      */
     protected function getContainerLoader(ContainerInterface $container)
     {
+        if (!$container instanceof ContainerBuilder) {
+            throw new \DomainException(sprintf('Container must be an instance of Symfony\Component\DependencyInjection\ContainerBuilder, got %s', get_class($container)));
```

Comment:
Not sure.. i tend to not couple context and domain. The message is about the container state, not whether it's the first, second or 10th argument.

It can be a DX improvement.. but to be honest i think when calling `getContainerLoader($myContainerVariableName)` and it throws `Container must be ...` is pretty straightforward ;-)